File: 646#MaximumLengthOfPairChain.go
// 646. Maximum Length of Pair Chain

package main
import "fmt"
import "sort"
import "math"

func findLongestChain(pairs [][]int) int {
    cur := math.MinInt32
    ans := 0
    sort.SliceStable(pairs, func(i, j int) bool {return pairs[i][1] < pairs[j][1]})
    for _, pair := range pairs {
        if cur < pair[0] {
            cur = pair[1]
            ans++
        }
    }
    return ans
}

File: 646#Maximum_Length_of_Pair_Chain.py
# 646. Maximum Length of Pair Chain

class Solution(object):
    def findLongestChain(self, pairs):
        cur, ans = float('-inf'), 0
        for x, y in sorted(pairs, key = operator.itemgetter(1)):
            if cur < x:
                cur = y
                ans += 1
        return ans